core/templates/dev/head/filtersSpec.js

```diff
@@ -483,6 +484,21 @@ describe('Testing filters', function() {
       'Text Input 3');
     }));
 
+  it('should remove extra new lines', inject(function($filter) {
+    var filter = $filter('removeExtraLines');
+
+    expect(filter('<p><br></p>')).toEqual('');
+    expect(filter('<p>abc</p>')).toEqual('<p>abc</p>');
+    expect(filter('<p>abc</p><p><br></p><p>abc</p>')
+    ).toEqual('<p>abc</p><p><br></p><p>abc</p>');
```

Comment:
Here and below, break up the line as so:

    expect(filter('<p>abc</p><p><br></p><p>abc</p>')).toEqual(
        '<p>abc</p><p><br></p><p>abc</p>');

Usually, you want to split the line at the opening delimiter rather than the closing one.